src/AzureReadStream.cs

```diff
@@ -134,7 +153,24 @@ public override int Read(byte[] buffer, int offset, int count)
                 DropSyncBuffer();
                 _position += count;
                 AzureRetry.Do(
-                    c => _blob.DownloadRangeToByteArrayAsync(buffer, offset, position, count, c),
+                    async c =>
+                    {
+                        var downloadUrl = _blob.GenerateSasUri(new BlobSasBuilder(BlobContainerSasPermissions.Read,
+                                                               new DateTimeOffset(DateTime.UtcNow.AddDays(1))));
+                        using (var httpClient = new HttpClient())
+                        {
+                            httpClient.DefaultRequestHeaders.Add("x-ms-range", $"bytes={position}-{offset + _position - 1}");
```

Comment:
Unless I'm mistaken `offset` is strictly related to `buffer`, not to the stream, so this should be `{position}-{position+count-1}` (let's avoid reliance on `_position` to be consistent).